backend/apps/volontulo/tests/views/api/test_contact.py

```diff
@@ -0,0 +1,158 @@
+        users[1].userprofile.save()
+        users[1].save()
+
+        res = self.client.get(ENDPOINT_URL)
+
+        self.assertDictEqual(res.data, {
+            'administrator_emails': [users[0].email, users[1].email],
+            'applicant_types': ContactSerializer.APPLICANT_CHOICES,
+        })
+
+
+class TestContactViewPOST(APITestCase, TransactionTestCase):
+
+    def setUp(self):
+        self.payload = {
+            'administrator_email': get_admin_email(),
+            'applicant_email': 'michal.kowalski@wp.pl',
+            'applicant_name': 'Michał Kowalski',
+            'applicant_type': ContactSerializer.VOLUNTEER,
+            'message': 'Chcę zaangażować się w jakąś akcję, ale nie wiem jak.'
+                       ' Możecie mi pomóc :)?',
+            'phone_no': '+ 00 48 600 700 800',
+        }
+
+    def test_missing_fields(self):
+        res = self.client.post(ENDPOINT_URL, {}, format='json')
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(res.data, {
+            'administrator_email': ['To pole jest wymagane.'],
+            'applicant_email': ['To pole jest wymagane.'],
+            'applicant_name': ['To pole jest wymagane.'],
+            'applicant_type': ['To pole jest wymagane.'],
+            'message': ['To pole jest wymagane.'],
+            'phone_no': ['To pole jest wymagane.'],
+        })
+
+    def test_min_length(self):
+        self.payload.update({
+            'applicant_name': 'K',
+            'message': 'Cześć!'
+        })
+        res = self.client.post(ENDPOINT_URL, self.payload, format='json')
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertDictEqual(res.data, {
+            'applicant_name': [
+                'Upewnij się, że pole ma co najmniej 3 znaków.',
+            ],
+            'message': [
+                'Upewnij się, że pole ma co najmniej 10 znaków.',
+            ],
+        })
+
+    def test_max_length(self):
+        self.payload.update({
+            'applicant_email': 'A' * 150 + '@googlemail.com',
+            'applicant_name': 'A' * 151,
+            'message': 'A' * 2001,
+            'phone_no': '6' * 21,
+        })
+        res = self.client.post(ENDPOINT_URL, self.payload, format='json')
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertDictEqual(res.data, {
+            'applicant_email': [
+                'Upewnij się, że to pole ma nie więcej niż 150 znaków.',
+            ],
+            'applicant_name': [
+                'Upewnij się, że to pole ma nie więcej niż 150 znaków.',
+            ],
+            'message': [
+                'Upewnij się, że to pole ma nie więcej niż 2000 znaków.',
+            ],
+            'phone_no': [
+                'Upewnij się, że to pole ma nie więcej niż 20 znaków.',
+            ],
+        })
+
+    def test_admin_email_validation(self):
+        self.payload.update({
```

Comment:
ditto

Reply:
Tests in class are executed one at a time, so there is no risk. I will refactor anyway.

Reply:
One at the time, that's true, but I doubt that anybody can ensure order of tests execution